clash-cores/src/Clash/Cores/Xilinx/Unisim/DnaPortE2/Internal.hs

```diff
@@ -0,0 +1,129 @@
+{-|
+Copyright   :  (C) 2022-2024, Google Inc.
+License     :  BSD2 (see the file LICENSE)
+Maintainer  :  QBayLogic B.V. <devops@qbaylogic.com>
+-}
+
+{-# OPTIONS_HADDOCK hide #-}
+
+module Clash.Cores.Xilinx.Unisim.DnaPortE2.Internal where
+
+import Clash.Cores.Xilinx.Xpm.Cdc.Internal
+import Clash.Explicit.Prelude
+
+data State
+  = Load
+  -- ^ Load the DNA register with an identifier unique to the FPGA
+  | Shift (BitVector 96)
+  -- ^ Shift out a bit from the DNA register
+  | Done (BitVector 96)
+  -- ^ Valid DNA shifted out
+  deriving (Generic, NFDataX)
+
+type DIN = Bit
+type DOUT = Bit
+type READ = Bool
+type SHIFT = Bool
```

Comment:
Why is it all caps?

Reply:
Probably because it was in the datasheet. Will change!